### ISSUE TITLE
On GraphDB add v1.0 to the REST API path

### ISSUE THREAD

**Issue by DinisCruz** [TeamMentor/TM_4_0_Design#289]
Add versioning information to the TM_Graph url so that it is future proof

At the moment the API calls look like:
* http://localhost:1332/git/status/
* http://localhost:1332/graph-db/contents/
* http://localhost:1332/graph-db/query/Logging
* http://localhost:1332/graph-db/queries

and they should be 
* http://localhost:1332/v1.0/git/status/
* http://localhost:1332/v1.0/graph-db/contents/
* http://localhost:1332/v1.0/graph-db/query/Logging
* http://localhost:1332/v1.0/graph-db/queries

@tekgirl77 can you write the test that represents the issue? (which is basically a GET call to ```http://localhost:1332/git/status/``` and confirm that we get a valid response)

This test should be placed in the https://github.com/TeamMentor/TM_4_0_GraphDB tests

**Comment by roman87** [comment]
Personally I don't like the URL proposed. First we should indicate that this is an API we are talking to so say https://localhost:1332/a/
then version, we don't need a v and a (.) in there  https://localhost:1332/a/01/git/status 
But thas my 5c. anyhow this is something that we will not be doing until sprint 3.

**Comment by DinisCruz** [comment]
what about https://localhost:1332/api/v1/git/status  ?

that is similar to what GitHub does https://developer.github.com/v3/#current-version

**Comment by roman87** [comment]
ok that works. hopefully we will not run out of single digits any time soon :)

**Comment by tekgirl77** [comment]
@DinisCruz and @roman87 I have drafted this code into the following file:

https://github.com/tekgirl77/TM_4_0_GraphDB/blob/Issue_289_Version_Graph_URLs/test/_issues/Version-Check.test.coffee

I wasn't sure where it should live permanently so for now it is in the /tests/_issues/ directory.


Do we want to add all of the target pages to this unit test that are listed in the swagger api:
http://localhost:1332/docs/?url=http://localhost:1332/v1.0/api-docs

i.e.
/graph-db/subjects/
/graph-db/predicates/
/graph-db/objects/

And I presume we also want to version control the other pages and include those as well - 
i.e.
/data/Library
/data/queries/
/git/log
/git/head

etc.

Please let me know if this looks good so far and I can continue adding additional graph URLs.

**Comment by DinisCruz** [comment]
well we need to have tests for all those methods, but we don't need to have them all in this issue 

Some of those APIs can have a impact on the current test DB, which is why in some cases the test needs to create a separate test environment. for example [this test](https://github.com/TeamMentor/TM_4_0_GraphDB/blob/master/test/api/Config-API.test.coffee#L80-L92):

```coffee
  it 'delete_data_cache', (done)->
        tmp_Cache_Root = '.tmp_Cache_Folder'
        using configApi.cache, ->
          @._cacheFolder = tmp_Cache_Root
          @.cacheFolder().folder_Create().assert_Folder_Exists()
          @.cacheFolder().path_Combine('test_file.txt').file_Create('aaaa')
          @.cacheFolder().files().assert_Not_Empty()
          clientApi.delete_data_cache (data)=>
            @.cacheFolder().files().assert_Empty()
            data.obj.assert_Is "deleted all files from folder #{@.cacheFolder()}"
            tmp_Cache_Root.folder_Delete_Recursive()
            tmp_Cache_Root.assert_Folder_Not_Exists()
            done()
```

In terms of the issue you are testing (#289), what you want to do is to make the actual test (for example https://github.com/tekgirl77/TM_4_0_GraphDB/blob/Issue_289_Version_Graph_URLs/test/_issues/Version-Check.test.coffee#L32) into something like https://github.com/TeamMentor/TM_4_0_QA/blob/master/tests/regression-sprint-1.coffee#L10

In this case what you are describing is the fact that the url ```/git/status``` is working, when it really should be ```api/v1/git/status```

**Comment by tekgirl77** [comment]
Okay... base on your initial comment above, I was thinking you wanted the tests for #289 to validate the current links are working the way they are now - a GET call to confirm a valid response?

Are you saying to write the tests to break now, for example, to be written as:
href.assert_Is '/api/v1/git/status'
href.assert_Is_Not '/git/status'

I have to hook into the api's in order to GET the graph pages, correct?  Otherwise I was getting 404 errors.  

Thanks for the help, just trying to make sure I understand what you want!  :)

**Comment by DinisCruz** [comment]
> Are you saying to write the tests to break now
yes and you only need a couple cases (ideally a couple that run fast with server side data processing)

> I have to hook into the api's in order to GET the graph pages, correct?
yes you will need to have that initial set up (like you are doing now)

**Comment by tekgirl77** [comment]
Got it, thanks @DinisCruz !
